app/src/main/java/com/lagradost/cloudstream3/ui/account/AccountDialog.kt

```diff
@@ -48,11 +52,32 @@ object AccountDialog {
                 callback.invoke(null)
             }
             .setOnDismissListener {
-                if (binding.pinEditTextError.isVisible()) {
+                if (binding.pinEditTextError.isVisible) {
                     callback.invoke(null)
                 }
             }
-            .create()
+
+        if (isNewPin) {
+            if (errorText != null) binding.pinEditTextError.visibleWithText(errorText)
+            builder.setPositiveButton(R.string.setup_done) { _, _ ->
+                if (binding.pinEditTextError.isVisible) {
```

Comment:
While this works, it is really bad to confuse state with the user interface. In a perfect world the UI state should be entirely separate from the business logic of the app, and I do know that this app does not entirely follow this at all times, this is too much. I think I failed to notice this before, but going forward you should use variables or view models to do this

Reply:
Dont do any sort of control flow operations based on the look of UI

Reply:
I think this is fixed